From e5fd143f215f072404c544f694cb026a4231503e Mon Sep 17 00:00:00 2001
From: Chen Qi <Qi.Chen@windriver.com>
Date: Thu, 21 Feb 2019 16:23:24 +0800
Subject: [PATCH 01/26] binfmt: Don't install dependency links at install time
 for the binfmt services

use [Install] blocks so that they get created when the service is enabled
like a traditional service.

The [Install] blocks were rejected upstream as they don't have a way to
"enable" it on install without static symlinks which can't be disabled,
only masked. We however can do that in a postinst.

Upstream-Status: Denied

Signed-off-by: Ross Burton <ross.burton@intel.com>
Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
[rebased for systemd 243]
Signed-off-by: Scott Murray <scott.murray@konsulko.com>
---
 units/meson.build                       | 2 --
 units/proc-sys-fs-binfmt_misc.automount | 3 +++
 units/systemd-binfmt.service.in         | 4 ++++
 3 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/units/meson.build b/units/meson.build
index 96f4852741..0a3a4fee67 100644
--- a/units/meson.build
+++ b/units/meson.build
@@ -156,7 +156,6 @@ units = [
         {
           'file' : 'proc-sys-fs-binfmt_misc.automount',
           'conditions' : ['ENABLE_BINFMT'],
-          'symlinks' : ['sysinit.target.wants/'],
         },
         {
           'file' : 'proc-sys-fs-binfmt_misc.mount',
@@ -258,7 +257,6 @@ units = [
         {
           'file' : 'systemd-binfmt.service.in',
           'conditions' : ['ENABLE_BINFMT'],
-          'symlinks' : ['sysinit.target.wants/'],
         },
         {
           'file' : 'systemd-bless-boot.service.in',
diff --git a/units/proc-sys-fs-binfmt_misc.automount b/units/proc-sys-fs-binfmt_misc.automount
index 7ec21e76c9..fee4d1345f 100644
--- a/units/proc-sys-fs-binfmt_misc.automount
+++ b/units/proc-sys-fs-binfmt_misc.automount
@@ -22,3 +22,6 @@ Before=shutdown.target
 
 [Automount]
 Where=/proc/sys/fs/binfmt_misc
+
+[Install]
+WantedBy=sysinit.target
diff --git a/units/systemd-binfmt.service.in b/units/systemd-binfmt.service.in
index 318bf8efc2..6ef684861d 100644
--- a/units/systemd-binfmt.service.in
+++ b/units/systemd-binfmt.service.in
@@ -14,6 +14,7 @@ Documentation=https://docs.kernel.org/admin-guide/binfmt-misc.html
 Documentation=https://systemd.io/API_FILE_SYSTEMS
 DefaultDependencies=no
 Conflicts=shutdown.target
+Wants=proc-sys-fs-binfmt_misc.automount
 After=proc-sys-fs-binfmt_misc.automount
 After=proc-sys-fs-binfmt_misc.mount
 After=local-fs.target
@@ -31,3 +32,6 @@ RemainAfterExit=yes
 ExecStart={{LIBEXECDIR}}/systemd-binfmt
 ExecStop={{LIBEXECDIR}}/systemd-binfmt --unregister
 TimeoutSec=90s
+
+[Install]
+WantedBy=sysinit.target
-- 
2.34.1

